### ISSUE TITLE
How to export custom YOLOv8 classification model with INT8 calibration

### ISSUE THREAD

**Issue by pornpra** [ultralytics/ultralytics#14732]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Currently, I am using Ultralytics version 8.2.18 and have tried to convert a custom YOLOv8 classification model from FP32 to INT8 using this command

`yolo export model=yolov8s_car_brand_fp32_25_07_2024.pt format=engine device=0 int8=True workspace=4 data=car_brand.yaml `

, but I got this error:

![352481988-f517521c-21c5-40d6-ad61-70bc7be7946c](https://github.com/user-attachments/assets/e1c2c96b-a27f-445a-8fd4-068c608bd053)

Inside car_brand.yaml, I have already defined the path and list of image files in .txt format for both the training set and test set as follows:

path: /home/nvidia/ORDS/int8/car_brand/yolov8
train: train_car_brand.txt
val: test_car_brand.txt

I am not sure how I should define content inside my custom yaml. Can you help me solve this export issue?

**Comment by pderrenger** [comment]
@pornpra to export your custom YOLOv8 classification model with INT8 calibration, ensure your dataset YAML file is correctly formatted and that the paths are accessible. The error might be due to incorrect paths or formatting issues in your `car_brand.yaml`. Verify that the paths to your training and validation datasets are correct and that the dataset files are properly formatted.

Additionally, make sure you are using the latest version of the Ultralytics package, as updates often include bug fixes and improvements. If the issue persists, please provide more details about the error message for further assistance. For more information on exporting models with TensorRT and INT8 calibration, you can refer to the [Ultralytics documentation](https://docs.ultralytics.com/integrations/tensorrt/).

**Comment by Y-T-G** [comment]
That's not the correct dataset format for classification. Classification doesn't use yaml.

**Comment by pornpra** [comment]
@Y-T-G 

Can you share the correct dataset format for classification ?

**Comment by Y-T-G** [comment]
https://docs.ultralytics.com/datasets/classify/#train

**Comment by pornpra** [comment]
@pderrenger 

I updated Ultralytics to version 8.2.66, but I still get the same error. I'm not sure if the latest Ultralytics version supports converting custom YOLOv8 classification models from FP32 to INT8. If it does, please provide a clear solution or reference.

**Comment by pderrenger** [comment]
Thank you for updating to version 8.2.66. The latest version does support converting custom YOLOv8 classification models from FP32 to INT8. Ensure your dataset is correctly formatted for classification, which typically involves organizing your images into subdirectories named after their classes. If the issue persists, please verify that your dataset paths are correct and accessible. For detailed guidance, refer to the [Ultralytics classification dataset documentation](https://docs.ultralytics.com/datasets/classify/#train). If you continue to face issues, please share the specific error message for further assistance.

**Comment by Egorundel** [comment]
@pornpra 

Hello!
How did you create the calibration_data.cache file?

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #22192: Verify passed dataset is a directory and not a file for classification training](https://github.com/ultralytics/ultralytics/pull/22192) by @Y-T-G with contributions from @glenn-jocher.

What changed and why this may resolve the issue:
- The classification dataset checker now validates that the provided path is a directory.
- If a file-like path is given (e.g., has a suffix), a clear ValueError is raised with guidance instead of attempting an auto-download.
- This avoids confusing behavior and helps ensure classification training uses the expected directory structure.
- See the [Classification dataset docs](https://docs.ultralytics.com/datasets/classify/) for the required layout.

How to test the fix:
- Install from the latest main branch:
  - `pip install -U "git+https://github.com/ultralytics/ultralytics.git@main"`
- Retry your workflow, ensuring the classification dataset path points to a directory (containing subfolders like /train and /val). Example:
  ```python
  from ultralytics import YOLO

  model = YOLO('yolo11n-cls.pt')
  model.train(data='path/to/classify_dataset')  # directory with /train and /val
  ```
- Alternatively, wait for the next official PyPI release and then:
  - `pip install -U ultralytics`

Feedback requested:
- Please let us know if the merged changes resolve the issue in your environment.
- If the problem persists, share a minimal snippet and the exact dataset path you’re passing (redact sensitive info).

Thanks for reporting this 🙏 and feel free to open follow-ups with any additional details.
